### PR TITLE
[dagster-ibis] Generate I/O manager implementation

### PR DESCRIPTION
**Pull request by deepyaman**
## Summary & Motivation

Create a minimal Ibis I/O manager, implementing the proposal from https://www.notion.so/dagster/RFC-Universal-I-O-manager-1dc18b92e462804093b9e7a136cdf9e2.

This is intended as PoC for feedback; there are probably some edge cases that may be rough (e.g. if you have `table_slice.partition_dimensions and len(context.asset_partition_keys) == 0` in `load_input()`, the empty table may not be properly handled).

The `delete_table_slice()` implementation is a bit hacky/could be "productionized." This can be a separate, broader improvement across I/O managers, too; technically, using `UPSERT`/`MERGE` functionality makes more sense than deleting and inserting (there is an issue internally on at least wrapping the delete and insert in a transaction, but a single operation would be cleaner), which could be enabled pretty cleanly through https://github.com/ibis-project/ibis/issues/5391.

## How I Tested These Changes

BK

I've tested the I/O manager with assets and ops, as well as on static- and time-partitioned data. I haven't made a significant effort to test edge cases and tangential functionality, like the metadata that gets written.

## Changelog

Added `dagster-ibis` and the `IbisIOManager`.

### CODE REVIEW

**Comment by deepyaman** [comment]
* **#29901** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29901?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29901?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-airbyte/tox.ini [hunk 1]
```diff
@@ -13,8 +13,7 @@ install_command = /bin/bash {env:DAGSTER_GIT_REPO_DIR}/scripts/uv-retry-install.
 deps =
   -e ../../dagster[test]
   -e ../../dagster-pipes
-  -e ../../libraries/dagster-shared
-  -e ../../dagster-test
+  -e ../dagster-shared
```
Comment (deepyaman) [inline python_modules/libraries/dagster-airbyte/tox.ini:16]:
```suggestion
  -e ../dagster-shared
  -e ../../dagster-test
```
